README.md

```diff
@@ -0,0 +1,219 @@
+        1,                      // amount of tokens to mint
+        { value: pricePerMint } // price per token
+    );
+```
+
+She was inspired by different tastes and she created 2 recipes adding them **directly** to the cookbook.
+
+```typescript
+const INITIAL_ALICE_recipeS = 2;
+    for (let i = 0; i < INITIAL_ALICE_recipeS; i++) {
+        const recipe_ID = actual_recipe_id;
+        actual_recipe_id++;    // Update to the next ID
+        await cookBookInstance
+            .connect(ALICE)
+            .addAssetEntry(     // add a new asset to the collection (not binded to any token)
+                `ipsf://metadata/master_recipe_${recipe_ID}`  // metadata URI of the asset
+            );
+        await cookBookInstance.
+            connect(ALICE).
+            addAssetToToken(
+                ALICE_BOOK_ID,  // ID of the token that will receive the asset
+                recipe_ID,     // ID of the asset to add
+                0               // ID of the asset to replace with the new one. 0 == none
+            );
+    }
+```
+
+## recipe improvement - Asset replacement
+
+During his travel Master created a recipe dedicated to his student and he decided to **send** it to Alice, to add it to her cookbook.
+
+```Typescript
+// Master creates a new recipe and Alice adds it to her cookbook
+    await cookBookInstance
+        .connect(MASTER)
+        .addAssetEntry(     // add a new asset to the collection (not binded to any token)
+            `ipsf://metadata/master_recipe_${actual_recipe_id}`   // metadata URI of the asset
+        );
+```
+
+Alice **accepted** suddently the recipe gifted from her master...
+
+```Typescript
+await cookBookInstance.connect(ALICE).
+        addAssetToToken(
+            ALICE_BOOK_ID,      // ID of the token that will receive the asset
+            actual_recipe_id,  // ID of the asset to add
+            0                   // ID of the asset to replace with the new one. 0 == none
+        );
+    actual_recipe_id++;    // Update to the next ID
+```
+
+...but she discovered that the recipe was missing something. :confused:
+After a long searching process and many attempts she found the right missing ingredient.
+So Alice fixed the recipe and she replaced the old one with this new improved one. :pencil:
+
+```Typescript
+// Alice creates a recipe and replace the latest added with it
+    await cookBookInstance
+        .connect(ALICE)
+        .addAssetEntry(     // add a new asset to the collection (not binded to any token)
+            `ipsf://metadata/alice_recipe_${actual_recipe_id}`    // metadata URI of the asset
+        );
+    const recipe_TO_REPLACE = 6;
+    await cookBookInstance
+        .connect(ALICE)
+        .addAssetToToken(
+            ALICE_BOOK_ID,      // ID of the token that will receive the asset
+            actual_recipe_id,  // ID of the asset to add
+            recipe_TO_REPLACE  // ID of the asset to replace with the new one
```

Comment:
It'd be nice, totally optional, if we have recipes name.
So instead of "actual" (I think you meant "current"), you would have something like `original_apple_pie_id`.
and instead of `recipe_TO_REPLACE` (let's be consistent with casing) you could have `improved_apple_pie_id`

Reply:
I've replaced `recipe_TO_REPLACE ` with `WRONG_RECIPE_ID` and `actual_recipe_id ` with `FIXED_RECIPE_ID`. Is it ok?